memguard_test.go

```diff
@@ -521,3 +505,43 @@ func TestGetBytes(t *testing.T) {
 		t.Error("pointer does not describe actual memory")
 	}
 }
+
+func TestFinalizer(t *testing.T) {
+	b, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ib := b.lockedBuffer
+
+	c, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ic := c.lockedBuffer
+
+	runtime.GC()
+	for ib.IsDestroyed() != true {
+		runtime.Gosched()
+	}
+
+	if ib.IsDestroyed() != true {
+		t.Error("expected b to be destroyed")
+	}
+	if ic.IsDestroyed() != false {
+		t.Error("expected c to not be destroyed")
+	}
+
+	runtime.KeepAlive(c)
+
+	runtime.GC()
+	for ic.IsDestroyed() != true {
+		runtime.Gosched()
+	}
+
+	if ib.IsDestroyed() != true {
```

Comment:
In the current implementation, it can't. But tests are supposed to be used also to verify that future changes in the implementation don't modify the API. It was just for clarity. I can remove it.